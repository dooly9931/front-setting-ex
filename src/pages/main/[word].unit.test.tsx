import React from "react";
import { shallow } from "enzyme";
import MainWord from "./[word]";

import { getPage } from "next-page-tester";
import { fireEvent, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Router from "next/router";

describe("Main", () => {
  it("should render home page correctly", async () => {
    // server side testing
    const { render } = await getPage({ route: "/main/test" });
    render();

    await waitFor(() => {
      expect(screen.getByText("Main page")).toBeInTheDocument();
      expect(screen.getByText('URL argument: "test"')).toBeInTheDocument();
      expect(screen.getByText('URL argument - backend response: "testtesttest"')).toBeInTheDocument();
    });
    // fireEvent(screen.getByText("home page"), new MouseEvent("click"));
  });

  it("should handle home page click", async () => {
    // client side testing
    const spyRouterPush = jest
      .spyOn(Router, "push")
      .mockImplementation((url) => new Promise((resolve, reject) => true));
    const component = shallow(<MainWord word="test2" word3="test2test2test2" randomHex="0x0000" />);
    const buttonWrapper = component.find("button");
    buttonWrapper.at(1).simulate("click");
    expect(spyRouterPush).toHaveBeenCalledTimes(1);
    expect(Router.push).toHaveBeenCalledWith("/");
    expect(component).toMatchSnapshot();
  });
});
