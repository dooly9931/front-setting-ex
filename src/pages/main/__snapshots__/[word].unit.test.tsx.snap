// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should handle home page click 1`] = `
<div
  id="main-page"
>
  <div>
    <div
      className="title-25 primary-orange"
    >
      Main page
    </div>
    <div
      className="subtitle-16 black"
    >
      Page Introduction: this is ted's main page
    </div>
    <button
      id="wallet-connect"
      type="button"
    >
      Connect Wallet
    </button>
  </div>
  <div>
    <div
      className="subtitle-16"
    >
      Web3 random number: 
    </div>
    <div
      className="subtitle-16"
    >
      URL argument: "
      test2
      "
    </div>
    <div
      className="subtitle-16"
    >
      URL argument - backend response: "
      test2test2test2
      "
    </div>
    <div
      className="subtitle-16"
    >
      Web3 Random Hex: 
      0x0000
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      home page
    </button>
  </div>
</div>
`;
